Frameworks/Foundation/_NSInvocation.x86.h

```diff
@@ -0,0 +1,53 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#import <Starboard/SmartTypes.h>
+#import <Foundation/NSMethodSignature.h>
+
+#import <cstdint>
+#import <algorithm>
+#import <vector>
+
+// It is a _NSInvocationCallFrame's job to carve out allocation extents from a buffer.
+class _NSInvocationCallFrame {
+private:
+    StrongId<NSMethodSignature> _methodSignature;
```

Comment:
>StrongId<NSMethodSignature> [](start = 4, length = 27)

nit: ...whereas an extra retain/release on this copy probably isn't necessary since _NSInvocationCallFrame can never outlive its parent NSInvocation